spd/run_spd.py

```diff
@@ -141,23 +171,26 @@ def optimize(
             group["lr"] = step_lr
 
         microbatch_log_data = defaultdict[str, float](float)
-        for _ in range(config.gradient_accumulation_steps):
+        for _ in range(gradient_accumulation_steps):
             batch = extract_batch_data(next(train_iterator)).to(device)
 
-            target_out, pre_weight_acts = model.forward_with_pre_forward_cache_hooks(
-                batch, module_names=model.target_module_paths
+            # Use component_model for special forward pass (DDP wrapper doesn't have this method)
+            target_out, pre_weight_acts = component_model.forward_with_pre_forward_cache_hooks(
```

Comment:
TODO: This doesn't work with DistributedDataParallel, I think we're going to need to use `__call__` for everything and have an argument in ComponentModel.forward() indicating what time of forward pass to run (pre_cache, components, target, etc.).

Reply:
Done.